sdk/core/System.ClientModel/tests/gen.unit/ContextGeneratorTests.cs

```diff
@@ -1140,5 +1140,456 @@ public void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+#pragma warning restore TEST002 // Experimental type
+  public partial class LocalContext : ModelReaderWriterContext { }
+
+  [Experimental("TEST001")]
+  public class JsonModel : IJsonModel<JsonModel>
+  {
+      JsonModel IJsonModel<JsonModel>.Create(ref System.Text.Json.Utf8JsonReader reader, ModelReaderWriterOptions options) => new JsonModel();
+      JsonModel IPersistableModel<JsonModel>.Create(BinaryData data, ModelReaderWriterOptions options) => new JsonModel();
+      string IPersistableModel<JsonModel>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+      void IJsonModel<JsonModel>.Write(System.Text.Json.Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+      BinaryData IPersistableModel<JsonModel>.Write(ModelReaderWriterOptions options) => BinaryData.Empty;
+  }
+
+  [Experimental("TEST002")]
+  public class OtherModel : IJsonModel<OtherModel>
+  {
+      OtherModel IJsonModel<OtherModel>.Create(ref System.Text.Json.Utf8JsonReader reader, ModelReaderWriterOptions options) => new OtherModel();
+      OtherModel IPersistableModel<OtherModel>.Create(BinaryData data, ModelReaderWriterOptions options) => new OtherModel();
+      string IPersistableModel<OtherModel>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+      void IJsonModel<OtherModel>.Write(System.Text.Json.Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+      BinaryData IPersistableModel<OtherModel>.Write(ModelReaderWriterOptions options) => BinaryData.Empty;
+  }
+}
+""";
+            Compilation compilation = CompilationHelper.CreateCompilation(source);
+
+            var result =
+                CompilationHelper.RunSourceGenerator(compilation, out var newCompilation, out var generatedSources);
+            Assert.IsNotNull(result.GenerationSpec);
+
+            // Verify the new compilation still has warnings for direct usage but not for generated code
+            var diagnostics = newCompilation.GetDiagnostics();
+            var errors = diagnostics.Where(d => d.Severity == DiagnosticSeverity.Error).ToArray();
+            Assert.AreEqual(0, errors.Length, "Compilation should not have errors");
+
+            // Check for experimental warnings - should still have them for direct usage in TestUsage class
+            var experimentalWarnings = diagnostics
+                .Where(d => (d.Id == "TEST001" || d.Id == "TEST002") && !d.Location.IsInSource).ToArray();
+            Assert.AreEqual(0, experimentalWarnings.Length, "Generated code should not produce experimental warnings");
+
+            // Check the context file for pragma suppressions in the constructor
+            var contextSource = generatedSources.First(s => s.HintName == "LocalContext.g.cs");
+            var contextText = contextSource.SourceText.ToString();
+
+            // The constructor should have pragma suppressions for experimental types
+            StringAssert.Contains("#pragma warning disable TEST001", contextText);
```

Comment:
I wonder if we should assert lines 1207-1209 as one assert so we know they are in order together and not just they exist somewhere but not together?

*Note* multiple instances of this but only marking this one.